src/main/ruby/core/truffle/exception_operations.rb

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+module Truffle
+  module ExceptionOperations
+    def self.class_name(exception)
```

Comment:
Could you pass the receiver to this helper instead of the exception?
`class_name(exception)` looks like it would get the name of `exception.class`.